src/parts/AccountModal/index.tsx

```diff
@@ -98,82 +117,208 @@ const AccountModal = ({
     onClose();
   };
 
+  const handleAccountDisconnect = () => {
+    dispatch(changeAddressAction(''));
+    window.bridge.removeAccount();
+    onClose();
+  };
+
+  React.useEffect(() => {
+    // Sets selected wallet on modal open.
+    if (open) {
+      const selectedAccount = accounts.find(({ address: accountAddress }) => address === accountAddress);
+      if (selectedAccount) {
+        setSelectedWallet(selectedAccount.meta.source as WalletSourceName);
+      } else {
+        setSelectedWallet(undefined);
+      }
+    }
+  }, [address, accounts, open]);
+
+  // State of the modal content.
+  const contentState = (() => {
+    if (supportedWalletInstalled) {
+      if (selectedWallet === undefined) {
+        return AccountModalState.SELECT_WALLET;
+      } else {
+        if (accounts !== undefined && accounts.length > 0) {
+          return AccountModalState.SELECT_ACCOUNT;
+        } else {
+          return AccountModalState.NO_ACCOUNT_FOUND;
+        }
+      }
+    } else {
+      return AccountModalState.NO_WALLET_FOUND;
+    }
+  })();
+
   const renderContent = () => {
-    if (extensionWalletAvailable) {
+    switch (contentState) {
```

Comment:
One suggestion, I wonder if we could create some sub render functions for readability.

Reply:
Yes, good suggestion, it definitely added a little readability :+1: